eng/common/pipelines/templates/jobs/archetype-spec-gen-sdk.yml

```diff
@@ -148,6 +151,14 @@ jobs:
           sdk_gen_info="$sdk_gen_info '${{ parameters.ConfigPath }}',"
         fi
 
+        if [ "$(Build.Reason)" = "PullRequest" ]; then
+          optional_params="$optional_params --n=$(System.PullRequest.PullRequestNumber)"
+          specPrUrl="${{ parameters.SpecRepoUrl }}/pull/$(System.PullRequest.PullRequestNumber)"
+          sdk_gen_info="$sdk_gen_info spec PR: $specPrUrl"
+          echo "Setting SDK repository to the forked repository under azure-sdk organization"
+          echo "##vso[task.setvariable variable=SdkRepoOwner]azure-sdk"
```

Comment:
Why are we only doing this for pull requests? Shouldn't we always push to azure-sdk fork?

Reply:
It was already implemented by `git-push-changes.yml`, see this [SDK PR](https://github.com/Azure/azure-sdk-for-java/pull/43910) being created.

Reply:
That is very confusing as it seems like you are intending to use it in this step as well. However you already disabled the PR if the build reason is PullRequest so how would this ever be used?